jbpm-services/jbpm-executor/src/main/java/org/jbpm/executor/commands/error/ProcessAutoAckErrorCommand.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.jbpm.executor.commands.error;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.persistence.EntityManager;
+
+import org.jbpm.runtime.manager.impl.jpa.ExecutionErrorInfo;
+
+/**
+ * Command that will auto acknowledge process instance errors 
+ * based on the last initial node instance when the error happened.
+ * 
+ * Following parameters are supported by this command:
+ * <ul>
+ *  <li>EmfName - name of entity manager factory to be used for queries (valid persistence unit name)</li>
+ *  <li>SingleRun - indicates if execution should be single run only (true|false)</li>
+ *  <li>NextRun - provides next execution time (valid time expression e.g. 1d, 5h, etc)</li>
+ * </ul>
+ */
+public class ProcessAutoAckErrorCommand extends AutoAckErrorCommand {
+
+    private static final String RULE = "Process instances that previously failed but now are in different nodes - meaning node where they were was already completed - or completed/aborted";
+    
+    @SuppressWarnings("unchecked")
+    @Override
+    protected List<ExecutionErrorInfo> findErrorsToAck(EntityManager em) {
+        List<ExecutionErrorInfo> errorsToAck = new ArrayList<>();
+        
+        String findProcessErrorsQuery = "select error from ExecutionErrorInfo error "
+                + "where error.acknowledged =:acknowledged "
+                + "and error.processInstanceId in (select pil.id from ProcessInstanceLog pil where status in (:status))";
+        
+        List<ExecutionErrorInfo> processErrorsToAck = em.createQuery(findProcessErrorsQuery)
+                .setParameter("acknowledged", false)
+                .setParameter("status", Arrays.asList(2, 3))
```

Comment:
Could we replace the 2, 3 with ProcessInstance.STATE_COMPLETED, ProcessInstance.STATE_ABORTED?